packages/sources/superstate/src/transport/nav.ts

```diff
@@ -0,0 +1,203 @@
+  // before finalizing and saving the last returned data to a cache.
+  async execute(fundId: number, retryCount = 0) {
+    const providerDataRequestedUnixMs = Date.now()
+    const apiResponse = await this.makeRequest(fundId)
+    const providerDataReceivedUnixMs = Date.now()
+
+    if (!apiResponse.data?.length) {
+      const response = {
+        errorMessage: `The data provider did not return any value for fundId: ${fundId}`,
+        statusCode: 502,
+        timestamps: {
+          providerDataRequestedUnixMs,
+          providerDataReceivedUnixMs,
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+      await this.responseCache.write(this.name, [{ params: { fundId }, response }])
+      return
+    }
+
+    const data = apiResponse.data[0]
+    const result = Number(data.net_asset_value)
+
+    // DP updates previous working day's price on the next working day at 9:09 AM ET
+    // If there is no fresh price data, we try to re-fetch the API until 10:30 AM ET
+    // Skips checks and re-running on weekends
+    if (this.shouldCheckForExpectedData()) {
+      // At this point we know that conditions are met, and we should check for stale data and retry if it's not updated.
+      // If the most recent update from DP is not from previous working day, the data might be stale
+      const expectedDate = getPreviousNonWeekendDay(TZ)
+      if (data.net_asset_value_date !== expectedDate) {
+        // We should retry fetching DP until we get fresh data or the time is after 10:30 AM ET
+        if (isBeforeTime('10:30:00', TZ)) {
+          logger.warn(
+            `Expected last update - ${expectedDate}, actual ${
+              data.net_asset_value_date
+            }. Retry attempt ${retryCount + 1}. Retrying after ${
+              this.settings.RETRY_INTERVAL_MS
+            } ms`,
+          )
+          retryCount++
+          setTimeout(() => this.execute(fundId, retryCount), this.settings.RETRY_INTERVAL_MS)
+          // We don't `return` here and let the value be stored in cache on purpose.
+          // This way the EA will respond with the latest value from DP (even though it's not the value that the EA expects),
+          // while it tries to get a fresh update.
+        } else {
+          logger.warn(
+            `Max retires reached. Expected update - ${expectedDate}. Latest update - ${data.net_asset_value_date}.`,
+          )
+        }
+      }
+    }
+
+    const response = {
+      data: {
+        result,
+      },
+      result,
+      statusCode: 200,
+      timestamps: {
+        providerDataRequestedUnixMs,
+        providerDataReceivedUnixMs,
+        providerIndicatedTimeUnixMs: undefined,
+      },
+    }
+    await this.responseCache.write(this.name, [{ params: { fundId }, response }])
+    return response
+  }
+
+  async makeRequest(fundId: number) {
+    const { startDate, endDate } = getStartingAndEndingDates(this.settings.LOOKBACK_DAYS)
```

Comment:
We should be good for now given the retry mechanism, long lookback, and scheduler, but I think we could run into some timezone issues here if we were to reuse this lib